sdks/go/pkg/beam/external.go

```diff
@@ -16,10 +16,144 @@
 package beam
 
 import (
+	"context"
+	"fmt"
+
 	"github.com/apache/beam/sdks/go/pkg/beam/core/graph"
+	"github.com/apache/beam/sdks/go/pkg/beam/core/runtime/graphx"
 	"github.com/apache/beam/sdks/go/pkg/beam/internal/errors"
+	jobpb "github.com/apache/beam/sdks/go/pkg/beam/model/jobmanagement_v1"
+	pipepb "github.com/apache/beam/sdks/go/pkg/beam/model/pipeline_v1"
+	"google.golang.org/grpc"
 )
 
+// ExternalTransform represents the cross-language transform in and out of the Pipeline as a MultiEdge and Expanded proto respectively
+type ExternalTransform struct {
```

Comment:
Per the other thread, please move these to parameters on CrossLanguage and TryCrossLanguage instead. Do not try to force in compatibility with the legacy External, it's OK for them to have two separate calls and paths.
By having them as a struct it's not clear what is required and what is not, and the compiler won't help the user by failing at compile time. 

An aside: The other issue here is you've mixed up user side parameters with internal implementation details, and made them part of the API surface. APIs are easiest to use when the user knows how to fill everything and what is required or not. The components and Expanded transform and requirements fields are not something that users would be filling in for example. Types are cheap. Make a new type instead of trying to reuse something that almost fits.

Reply:
The current version was never meant to be representative of what the final API would be. Only after discussing code organization was the API surface meant to be refined. 

Thanks for the new type idea! It'll make separation of concerns much clearer.